src/Equations/Fluid/IonChargedDiffusionStochasticBTerm.cpp

```diff
@@ -0,0 +1,110 @@
+#include "DREAM/Constants.hpp"
+#include "DREAM/Equations/Fluid/IonChargedAdvectionDiffusionTerm.hpp"
+#include "DREAM/Equations/Fluid/IonChargedDiffusionStochasticBTerm.hpp"
+#include "DREAM/Equations/Fluid/IonEquationTerm.hpp"
+#include "FVM/Equation/DiffusionTerm.hpp"
+#include "DREAM/IonHandler.hpp"
+//#include "DREAM/NotImplementedException.hpp"
+#include "FVM/Grid/Grid.hpp"
+
+
+using namespace DREAM;
+
+IonChargedDiffusionStochasticBTerm::IonChargedDiffusionStochasticBTerm(FVM::Grid *grid, IonHandler *ions, bool allocCoefficients,
+	const len_t iIon, FVM::Interpolator1D* dBOverB, MultiInterpolator1D* DrrHat, FVM::UnknownQuantityHandler *u
+	) : IonChargedAdvectionDiffusionTerm<FVM::DiffusionTerm>(grid, 
+	ions, iIon, allocCoefficients), dBOverB(dBOverB), DrrHat(DrrHat) {
+	
+    SetName("IonChargedDiffusionStochasticBTerm");
+
+    this->id_ni = u->GetUnknownID(OptionConstants::UQTY_ION_SPECIES);
+    AddUnknownForJacobian(u, id_ni);
+    
+    this->id_Wi = u->GetUnknownID(OptionConstants::UQTY_WI_ENER);
+    AddUnknownForJacobian(u, id_Wi);
+    
+    this->id_Ni = u->GetUnknownID(OptionConstants::UQTY_NI_DENS);
+    AddUnknownForJacobian(u, id_Ni);
+	
+	Allocate();
+}
+
+IonChargedDiffusionStochasticBTerm::~IonChargedDiffusionStochasticBTerm(){
+	Deallocate();
+}
+
+void IonChargedDiffusionStochasticBTerm::Allocate(){
+    Deallocate();
+    
+    len_t nzs=ions->GetNzs();
+
+    const len_t nr = this->grid->GetNr();
+
+	// Derivatives wrt ion (charge state) densities
+    this->dDrrdni = new real_t*[Zion];
+    for(len_t Z0=1; Z0<=Zion; Z0++)
+        this->dDrrdni[Z0-1] = new real_t[(nr+1)*nzs];
+        
+    // Derivatives wrt ion thermal energy (of this species only)
+    this->dDrrdWi = new real_t*[Zion];
+    for(len_t Z0=1; Z0<=Zion; Z0++)
+        this->dDrrdWi[Z0-1] = new real_t[(nr+1)];
+        
+    // Derivatives wrt the total ion density (of this species only)
+    this->dDrrdNi = new real_t*[Zion];
+    for(len_t Z0=1; Z0<=Zion; Z0++)
+        this->dDrrdNi[Z0-1] = new real_t[(nr+1)];
+}
+
+void IonChargedDiffusionStochasticBTerm::Deallocate(){
+    for(len_t Z0=1; Z0<=Zion; Z0++)
+        delete [] this->dDrrdni[Z0-1];
+    delete [] this->dDrrdni;
+    
+    for(len_t Z0=1; Z0<=Zion; Z0++)
+        delete [] this->dDrrdWi[Z0-1];
+    delete [] this->dDrrdWi;
+    
+    for(len_t Z0=1; Z0<=Zion; Z0++)
+        delete [] this->dDrrdNi[Z0-1];
+    delete [] this->dDrrdNi;
+}
+
+void IonChargedDiffusionStochasticBTerm::SetCoeffsAllCS(const real_t t){
+}
+
+void IonChargedDiffusionStochasticBTerm::SetDiffCoeffsAllCS(const real_t t){
+}
```

Comment:
What is the purpose of these methods?

Reply:
Aha, they are declared virtual and null in the parent class? Then this is fine. (If it's not necessary to implement the methods in the derived class, I typically let the parent class implement the methods like this however to not have to write the extra few lines in every child class)